diffsims/sims/diffraction_simulation.py

```diff
@@ -106,29 +178,77 @@ def direct_beam_mask(self):
     @property
     def coordinates(self):
         """ndarray : The coordinates of all unmasked points."""
-        if self._coordinates is None:
-            return None
         return self._coordinates[self.direct_beam_mask]
 
     @coordinates.setter
     def coordinates(self, coordinates):
-        self._coordinates = coordinates
+        self._coordinates[self.direct_beam_mask] = coordinates
 
     @property
     def intensities(self):
         """ndarray : The intensities of all unmasked points."""
-        if self._intensities is None:
-            return None
         return self._intensities[self.direct_beam_mask]
 
     @intensities.setter
     def intensities(self, intensities):
-        self._intensities = intensities
+        self._intensities[self.direct_beam_mask] = intensities
+
+    def _get_transformed_coordinates(
+        self, angle, center=(0, 0), mirrored=False, units="real"
+    ):
+        """
+        Get the coordinates of diffraction spots rotated, flipped or shifted
+        in-plane
+
+        Parameters
+        ----------
+        angle: float
+            Angle in degrees
+        center: 2-tuple of floats
+            Center of the patterns
+        mirrored: bool
+            Mirror accross the x axis
+        units: str
+            "real" or "pixel"
+
+        Returns
+        -------
+        coords_transformed: np.ndarray of shape (N, 3) or (N, 2)
+            transformed coordinates
+        """
+        if units == "real":
+            coords_transformed = self.coordinates.copy()
+        else:
+            coords_transformed = self.calibrated_coordinates.copy()
+        cx, cy = center
+        x = coords_transformed[:, 0]
+        y = coords_transformed[:, 1]
+        mirrored_factor = -1 if mirrored else 1
+        theta = mirrored_factor * np.arctan2(y, x) + np.deg2rad(angle)
+        rd = np.sqrt(x ** 2 + y ** 2)
+        coords_transformed[:, 0] = rd * np.cos(theta) + cx
+        coords_transformed[:, 1] = rd * np.sin(theta) + cy
+        return coords_transformed
+
+    def rotate_shift_coordinates(self, angle, center=(0, 0), mirrored=False):
+        """Translate, rotate or mirror the pattern"""
```

Comment:
The docstring must be added here to the public API method, not the private one. The `units` can be documented in the private method docstring, pointing to the public API docstring.